dialogue-core/src/main/java/com/palantir/dialogue/core/RetryingChannel.java

```diff
@@ -144,10 +145,10 @@
                 .channelName(channelName)
                 .reason("qosResponse")
                 .build();
-        this.retryDueToThrowable = DialogueClientMetrics.of(metrics)
+        this.retryDueToThrowable = throwable -> DialogueClientMetrics.of(metrics)
```

Comment:
DialogueClientMetrics should be created up front and reused.
```suggestion
        DialogueClientMetrics dialogueClientMetrics = DialogueClientMetrics.of(metrics);
        this.retryDueToThrowable = throwable -> dialogueClientMetrics
```